libraries/ms-common/inc/bootloader_can.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+// This module allows the sending and receiving of CAN messages in the bootloader's CAN ID format.
+// It allows registering a callback which will be called when a bootloader CAN message is received
+// with client controller board ID 0 (msg_id == 0).
+// Requires interrupts, gpio, the event queue, soft timers and CAN to be initialized.
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "status.h"
+
+// Initialize the module using CanStorage and CanSettings, without needing to specify device_id
+// device_id will be overwritten to be SYSTEM_CAN_DEVICE_BOOTLOADER
+StatusCode bootloader_can_init(CanStorage *storage, CanSettings *settings);
+
+// Transmits a CAN message with data of length len, and sets the message type = CAN_MSG_TYPE_ACK
+// if is_start_message is true, otherwise, type = CAN_MSG_TYPE_DATA

```

Comment:
Your doc comments in this header seem to be describing implementation details of the functions, rather than their purpose. The user of this module doesn't care exactly how the module formats a CAN ID - in fact, the entire purpose of the module is so that they don't *have* to care. If they wanted to know how the module indicates whether a message is a start message, they could read the implementation file or the design document.

So instead of talking about what `bootloader_can_transmit` does with the message type, say that it transmits a bootloader datagram message with the given data, from the given board id, and that it's a start message if `is_start_message` is true and a non-start message otherwise.